test/insertion.jl

```diff
@@ -1370,4 +1370,96 @@ end
     end
 end
 
+@testset "multi element append!/prepend!/push!/pushfirst!" begin
+    df = DataFrame(a=1, b=2)
+    @test append!(df) == DataFrame(a=1, b=2)
+    @test prepend!(df) == DataFrame(a=1, b=2)
+    @test push!(df) == DataFrame(a=1, b=2)
+    @test pushfirst!(df) == DataFrame(a=1, b=2)
+    @test_throws ArgumentError append!(df, cols=:x) == DataFrame(a=1, b=2)
+    @test_throws ArgumentError prepend!(df, cols=:x) == DataFrame(a=1, b=2)
+    @test_throws ArgumentError push!(df, cols=:x) == DataFrame(a=1, b=2)
+    @test_throws ArgumentError pushfirst!(df, cols=:x) == DataFrame(a=1, b=2)
+
+    for x in (DataFrame(a=3, b=4), (a=[3], b=[4]), [(a=3, b=4)]),
+        y in (DataFrame(a=5, b=6), (a=[5], b=[6]), [(a=5, b=6)]),
+        z in (DataFrame(a=7, b=8), (a=[7], b=[8]), [(a=7, b=8)])
+        @test append!(copy(df), x, y) ==
+              DataFrame(a=1:2:5, b=2:2:6)
+        @test append!(copy(df), x, y, z) ==
+              DataFrame(a=1:2:7, b=2:2:8)
+        @test prepend!(copy(df), x, y) ==
+              DataFrame(a=[3, 5, 1], b=[4, 6, 2])
+        @test prepend!(copy(df), x, y, z) ==
+              DataFrame(a=[3, 5, 7, 1], b=[4, 6, 8, 2])
+    end
+
+    for x in (DataFrame(a=3, b=4)[1, :], (a=3, b=4), (3, 4)),
+        y in (DataFrame(a=5, b=6)[1, :], (a=5, b=6), (5, 6)),
+        z in (DataFrame(a=7, b=8)[1, :], (a=7, b=8), (7, 8))
+        @test push!(copy(df), x, y) ==
+              DataFrame(a=1:2:5, b=2:2:6)
+        @test push!(copy(df), x, y, z) ==
+              DataFrame(a=1:2:7, b=2:2:8)
+        @test pushfirst!(copy(df), x, y) ==
+              DataFrame(a=[3, 5, 1], b=[4, 6, 2])
+        @test pushfirst!(copy(df), x, y, z) ==
+              DataFrame(a=[3, 5, 7, 1], b=[4, 6, 8, 2])
+        for cols in (:orderequal, :setequal)
```

Comment:
Also test cases that error?

Reply:
I already test them at the very bottom of the new tests (I cannot test them here as in some cases they are allowed - note that these tests test mixing of named and unnamed rows).

Reply:
I see tests with tuples, but nothing with named tuples or `DataFrameRow` (which is needed to check that order is correct).

Reply:
These tests are already done in old tests.